amideploy/infrastructure/main.tf

```diff
@@ -0,0 +1,159 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+    tls = {
+      source  = "hashicorp/tls"
+      version = "~> 4.0"
+    }
+  }
+
+  required_version = ">= 1.4.0"
+}
+
+provider "aws" {
+  region  = var.aws_region
+  profile = var.aws_profile
+}
+
+provider "tls" {}
+
+resource "tls_private_key" "airflow_server_private_key" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "aws_key_pair" "generated_airflow_server_key" {
+  key_name   = "auto-generated-key"
+  public_key = tls_private_key.airflow_server_private_key.public_key_openssh
+}
+
+output "airflow_server_private_key_pem" {
+  value     = tls_private_key.airflow_server_private_key.private_key_pem
+  sensitive = true
+}
+
+resource "aws_instance" "ami_connect_airflow_server" {
+  ami           = "ami-087f352c165340ea1"
+  instance_type = "t3.medium"
+  vpc_security_group_ids = [aws_security_group.airflow_server_sg.id]
+  key_name      = aws_key_pair.generated_airflow_server_key.key_name
+
+  tags = {
+    Name = var.ami_connect_tag
+  }
+}
+
+resource "aws_eip" "ami_connect_airflow_server_ip" {
```

Comment:
An Elastic IP address, a.k.a. a stable public IP address we can associate with the EC2 server.